mava/components/jax/updating/terminators.py

```diff
@@ -14,5 +14,92 @@
 # limitations under the License.
 
 """Terminator component for Mava systems."""
+import abc
+from typing import Any, Type
 
-# TODO (Dries)
+import launchpad as lp
+from chex import dataclass
+
+from mava.components.jax.component import Component
+from mava.core_jax import SystemParameterServer
+
+
+class Terminator(Component):
+    @abc.abstractmethod
+    def __init__(
+        self,
+        config: Any,
+    ):
+        """_summary_
+
+        Args:
+            config : _description_.
+        """
+        self.config = config
+
+    @abc.abstractmethod
+    def on_parameter_server_run_loop_termination(
+        self, parameter_sever: SystemParameterServer
+    ) -> None:
+        """_summary_"""
+        pass
+
+    @staticmethod
+    def name() -> str:
+        """_summary_
+
+        Returns:
+            _description_
+        """
+        return "temination_condition"
+
+
+@dataclass
+class ParameterTerminatorConfig:
+    parameter_termination_key: str = "executor_steps"
+    parameter_termination_value: int = 5000
+
+
+class ParameterTerminator(Terminator):
+    def __init__(
+        self,
+        config: Any = ParameterTerminatorConfig(),
+    ):
+        """_summary_
+
+        Args:
+            config : _description_.
+        """
+        self.config = config
+
+        valid_options = [
```

Comment:
Make dictionary instead of list

Reply:
No keys, so rather a set?

Reply:
Sorry, that's what I meant haha. i.e. `{item1, item2, item3}`